src/java/voldemort/store/socket/clientrequest/ClientRequestExecutor.java

```diff
@@ -235,6 +257,38 @@ protected void read(SelectionKey selectionKey) throws IOException {
     }
 
     @Override
+    protected void reportException(IOException e) {
+        clientRequest.reportException(e);
+    }
+
+    @Override
+    protected void connect(SelectionKey selectionKey) throws IOException {
```

Comment:
so you are moving the connect() effectively into NIO as well, instead of blocking the thread.. That seems to be essence of it? (If so +1, again one more thing that was always on the backburner.. super happy to see this getting traction).. 

NIO selector does not scale super linearly to number of connections, based on some experiments I ran a long time back. Just something to keep in mind, as pending connections piling up.. Client side may not be that much of deal technically.. pls let me know how it goes in prod..